homeassistant/components/upnp/__init__.py

```diff
@@ -209,26 +240,45 @@ class UpnpDataUpdateCoordinator(DataUpdateCoordinator):
     """Define an object to update data from UPNP device."""
 
     def __init__(
-        self, hass: HomeAssistant, device: Device, update_interval: timedelta
+        self,
+        hass: HomeAssistant,
+        device: Device,
+        device_entry: dr.DeviceEntry,
+        update_interval: timedelta,
     ) -> None:
         """Initialize."""
         self.device = device
+        self.device_entry = device_entry
 
         super().__init__(
-            hass, LOGGER, name=device.name, update_interval=update_interval
+            hass,
+            LOGGER,
+            name=device.name,
+            update_interval=update_interval,
+            update_method=self._async_fetch_data,
         )
 
-    async def _async_update_data(self) -> Mapping[str, Any]:
+    async def _async_fetch_data(self) -> Mapping[str, Any]:
         """Update data."""
-        update_values = await asyncio.gather(
-            self.device.async_get_traffic_data(),
-            self.device.async_get_status(),
-        )
+        try:
+            update_values = await asyncio.gather(
+                self.device.async_get_traffic_data(),
+                self.device.async_get_status(),
+            )
 
-        return {
-            **update_values[0],
-            **update_values[1],
-        }
+            return {
```

Comment:
Please only wrap the line that can raise in the try... except block. We can move the return statement to the end.